raiden/transfer/mediated_transfer/state.py

```diff
@@ -55,6 +55,26 @@ def __init__(self, our_address, transfer, routes, block_number, random_generator
         self.revealsecret = None
         self.canceled_transfers = list()
 
+    def __eq__(self, other):
+        if isinstance(other, InitiatorState):
```

Comment:
Do all the attributes have the comparison operator properly implemented? Also for the cases below?

One note. If we are to go with this "Compare Everything" approach and since the code is identical here and in the next 3 instances below I think it would be wise to abstract it into a utility function.